packages/compass-components/src/hooks/use-hotkeys.ts

```diff
@@ -0,0 +1,38 @@
+import { useMemo } from 'react';
+import { useHotkeys as useGlobalHotkeys } from 'react-hotkeys-hook';
+
+const isMac = () => navigator.userAgent.indexOf('Mac') !== -1;
+
+/**
+ *
+ * Modifier | macOS       | Windows/Linux
+ * -------- | ----------- | -------------
+ * ctrl     | control     | ctrl
+ * shift    | shift       | shift
+ * alt      | option      | alt
+ * meta     | command     | ctrl
+ *
+ */
+export const useHotkeys = (...args: Parameters<typeof useGlobalHotkeys>) => {
+  const [key] = args;
+  const ref = useGlobalHotkeys(...args);
+
+  const shortcut = useMemo(() => {
+    if (Array.isArray(key)) {
+      return key.map(mapKeyToShortcut);
+    }
+    return mapKeyToShortcut(key);
+  }, [key]);
+
+  return {
+    shortcut,
```

Comment:
Not a big deal, but can we avoid `useHotkeys` to also render the shortcut string? I feel that like this we would lose some flexibility. For example, I can only use the function that renders the shortcut consistently if I'm also binding it, which may not be the case: code mirror may register hotkeys that we may still want to document.

Could we have this rendering as a separate function or component?

Reply:
Yes we can avoid doing that. Actually this hook does not render it, it just maps it for rendering. With this approach we are forcing user to render what has been binded. Now that I am adding a component to render the shortcut, I'd still prefer to leave this here and return the shortcut (without mapping) and whenever user wants to render the shortcut, use the `Hotkey` component which will map and render it. Something like this:

```ts
const { shortcut } = useHotkeys('ctrl + a');

return (
	<Hotkey shortcut={shortcut} />
);
```

Alternatively, we can remove it from the hook and then let users render using `Hotkey` component. 

```ts
useHotkeys('ctrl + a');

return (
	<Hotkey shortcut="ctrl + a" />
);
```
Personally i like the former approach.